src/benchmarklib/benchmark_table_encoder.cpp

```diff
@@ -0,0 +1,143 @@
+#include "benchmark_table_encoder.hpp"
+
+#include "constant_mappings.hpp"
+#include "encoding_config.hpp"
+#include "resolve_type.hpp"
+#include "storage/base_encoded_segment.hpp"
+#include "storage/base_value_segment.hpp"
+#include "storage/chunk_encoder.hpp"
+#include "storage/table.hpp"
+#include "types.hpp"
+
+namespace {
+
+using namespace opossum;  // NOLINT
+
+SegmentEncodingSpec generate_segment_encoding_spec(const BaseValueSegment&) { return {EncodingType::Unencoded}; }
+
+SegmentEncodingSpec generate_segment_encoding_spec(const ReferenceSegment&) {
+  Fail("Did not expect a ReferenceSegment in base table");
+}
+
+SegmentEncodingSpec generate_segment_encoding_spec(const BaseEncodedSegment& base_encoded_segment) {
+  if (base_encoded_segment.compressed_vector_type()) {
+    switch (*base_encoded_segment.compressed_vector_type()) {
+      case CompressedVectorType::FixedSize1ByteAligned:
+      case CompressedVectorType::FixedSize2ByteAligned:
+      case CompressedVectorType::FixedSize4ByteAligned:
+        return {base_encoded_segment.encoding_type(), VectorCompressionType::FixedSizeByteAligned};
+      case CompressedVectorType::SimdBp128:
+        return {base_encoded_segment.encoding_type(), VectorCompressionType::SimdBp128};
+    }
+
+    Fail("GCC thinks this is reachable");
```

Comment:
At least in this case, we can have a better message: "Unexpected compressed vector type"

Reply:
If there really was a "CompressedVectorType" that we didn't handle in the switch above, then the compiler would complain about an enum member not being enumerated. I use the message "GCC thinks this is reachable" to mark places where Gcc - erroneously - thinks the control flow might reach, but it actually can't

Reply:
Are you using gcc 8? I thought most of those false positives were resolved

Reply:
I think this particular case was reported by the CI, I usually don't use GCC for local builds